packages/gitbook/src/lib/pages.ts

```diff
@@ -8,17 +8,68 @@ import {
 
 export type AncestorRevisionPage = RevisionPageDocument | RevisionPageGroup;
 
+type ResolvedPageDocument = {
```

Comment:
Haha I'm a pain in the ass, but this could be a nice type with a generic

```ts
type ResolvedPagePath<Page extends RevisionPageDocument | RevisionPageGroup> = {
  page: Page;
  ancestors: AncestorRevisionPage[];
}
```